### PR TITLE
feat(container-image): added container image service

### PR DESCRIPTION
**Pull request by ammmze**
Allows managing container images. I didn't see vyos documentation for the endpoint, but found the source code for it [here](https://github.com/vyos/vyos-1x/blob/b01a7a06ae63c2f0bbeb60a8cef567f4d333be30/src/services/vyos-http-api-server#L597-L626).

### TODO
- [x] update vyos rolling docker image, so integration tests can actually pass

### CODE REVIEW

**Comment by ammmze** [comment]
Looks like we might need a newer vyos rolling image for this. I think the one the integration tests run against doesn't have this `container-image` endpoint, so it appears to be getting HTML back instead of JSON.

```
invalid character '<' looking for beginning of value
```

Edit: I've adding some error handling in the "Request" method to check response status code, so we get a little more helpful error response than what we got before (shown above):

```
received non-successful (404) response from vyos api (https://localhost/container-image)
```

**Comment by foltik** [comment]
@ammmze Thanks so much for the contribution, this looks great!

I updated the vyos-rolling docker image to the latest version and ran CI on the dev branch with your patch merged, but looks like the integration tests are still failing. Here's the job in question: https://github.com/foltik/vyos-client-go/actions/runs/5329440777/jobs/9655118557. Have you seen this before?

**Comment by ammmze** [comment]
> @ammmze Thanks so much for the contribution, this looks great!
> 
> I updated the vyos-rolling docker image to the latest version and ran CI on the dev branch with your patch merged, but looks like the integration tests are still failing. Here's the job in question: https://github.com/foltik/vyos-client-go/actions/runs/5329440777/jobs/9655118557. Have you seen this before?

Excellent! Thank you for updating that. I'll dig into the integration tests in a bit and see if I can get them working.

**Comment by ammmze** [comment]
Still trying to figure out the issue. It looks like the vyos api is returning un-expected data. Normally the "data" would return something like:

```
REPOSITORY                TAG         IMAGE ID      CREATED      SIZE
docker.io/library/alpine  3.18.0      5e2b554c1c45  6 weeks ago  7.62 MB
```

And in my code changes, I am parsing that first line to ensure it matches a format like that and if it doesn't match, it will surface that error. The text in the parentheses ("sudo: unable to resolve host b88ac76194dd: System error") is the first line of the "data" text. So I don't think the container stuff is working correctly, or maybe it needs to be enabled first or something.

Edit: Ah...looks like it (container api) is working, we are just getting extra lines of data before that header. So the full "data" we are getting is:

```
sudo: unable to resolve host 62b1b7ccea6c: System error
REPOSITORY                TAG         IMAGE ID      CREATED       SIZE
docker.io/library/alpine  3.17.3      9ed4aefc74f6  2 months ago  7.34 MB
```

**Comment by ammmze** [comment]
There's still one more failing integration test. Getting a 400 response code during the `TestIntegration_Config_Delete`. Not sure why yet. I've gotta get rolling for normal work. I'll look more into it tonight.

**Comment by ammmze** [comment]
@foltik Got some updates in there and now the integration tests are all passing.

File: client/client_test.go [hunk 1]
```diff
@@ -117,15 +120,14 @@ func TestIntegration_Config_ShowRoot(t *testing.T) {
 	}
 }
 
-
 func TestIntegration_Config_SetValue(t *testing.T) {
 	client, ctx := make_client(t)
-	err := client.Config.Set(ctx, "system ntp listen-address", "1.2.3.4")
+	err := client.Config.Set(ctx, "service ntp listen-address", "1.2.3.4")
```
Comment (ammmze) [inline client/client_test.go]: Looks like newer vyos version puts `ntp` under `service` instead of `system`.

**Comment by foltik** [comment]
Looks good, thanks again for the contribution!

Rebased to main + added a commit with some tweaks + additional tests.

**Pull request merged**